docs/subnets/hello-world-precompile-tutorial.md

```diff
@@ -0,0 +1,976 @@
+
+### Prerequisites
+
+- Git Clone the [Subnet-EVM](https://github.com/ava-labs/subnet-evm) repo.
+- Git Clone [Avalanchego](https://github.com/ava-labs/avalanchego) repo.
+- Install [Avalanche Network Runner](https://docs.avax.network/subnets/network-runner)
+- Install [solc](https://github.com/ethereum/solc-js#usage-on-the-command-line)
+
+First install the latest version of Go. Follow the instructions [here](https://go.dev/doc/install). You can verify by running go version.
+
+Set `$GOPATH` environment variable properly for Go to look for Go Workspaces. Please read [this](https://go.dev/doc/gopath_code) for details. You can verify by running echo $GOPATH.
+
+As a few things will be installed into $`GOPATH/bin`, please make sure that `$GOPATH/bin`is in your `$PATH`, otherwise, you may get error running the commands below.
+
+Download the following prerequisites into your `$GOPATH`:
+
+```
+cd $GOPATH
+mkdir -p src/github.com/ava-labs
+cd src/github.com/ava-labs
+git clone git@github.com:ava-labs/subnet-evm.git
+git clone git@github.com:ava-labs/avalanchego.git
+git clone https://github.com/ava-labs/avalanche-network-runner.git
+npm install -g solc
+```
+
+## Tutorial
+
+For the tutorial, we will be working in a new branch in Subnet-EVM.
+
+```bash
+cd $GOPATH/src/github.com/ava-labs/subnet-evm
+git checkout -b hello-world-stateful-precompile
+```
+
+There will be code snippets to copy and paste into the appropriate files.
+
+We will first start off by creating the Solidity interface that we want our precompile to implement. This will be the HelloWorld Interface. It will have two simple functions, `sayHello()` and `setGreeting()`. These two functions will demonstrate the getting and setting respectively of a value using state access.
+
+We will place the interface in `./contract-examples/contracts`. You can copy and paste the below code into `./contract-examples/contracts/IHelloWorld.sol`.
+
+```bash
+cd contract-examples/contracts
+```
+
+```sol
+// (c) 2022-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+// SPDX-License-Identifier: MIT
+
+pragma solidity >=0.8.0;
+
+interface IHelloWorld {
+ // sayHello returns the string located at [key]
+  function sayHello() external returns (string calldata result);
+
+// setGreeting sets the string located at [key]
+  function setGreeting(string calldata response) external;
+}
+```
+
+Now we have an interface that our precompile can implement!
+Let's create an [abi](https://docs.soliditylang.org/en/v0.8.13/abi-spec.html#:~:text=Contract%20ABI%20Specification-,Basic%20Design,as%20described%20in%20this%20specification.) of our solidity code.
+
+In the same `./contract-examples/contracts` directory, let's run
+
+```
+solcjs --abi IHelloWorld.sol
+```
+
+This spits out the abi code in `./contract-examples/contracts`!
+Rename this file to be called `IHelloWorld.abi`
```

Comment:
Why do we have to rename, instead of keeping what is generated as `IHelloWorld_sol_IHelloWorld.abi`? Is it because you have to keep the tutorial working with your new changes?

Reply:
It's easier to read if this is called IHelloWorld.abi. This file name is not easy to follow. `IHelloWorld_sol_IHelloWorld.abi`

Reply:
I am doing a passthrough of the tutorial again today and will make the changes accordingly.

Reply:
> It's easier to read if this is called IHelloWorld.abi. This file name is not easy to follow. `IHelloWorld_sol_IHelloWorld.abi`

I agree with the name not easy to follow. I disagree that we should change filename which is an extra step that user has to do.

Reply:
I am fixing to reduce steps.